[Gong Chen Jun] iP

0x787af25e opened the pull request:
# A heading

A bullet list
- bullet point 1
- bullet point 2 
- bullet point 3

A numbered list
1. entry 1
2. entry 2
3. entry 3

A fenced code block
```sql
DROP TABLE pull_requests;
```
A task list
- [x] lorem ipsum
- [ ] dolor sit amet

An emoji🥲🥲🥲 (actually 3 of them)

> A blockquote 

[A hyperlink (this is actually the kind of music I listen to, maybe I really am insane](https://www.youtube.com/watch?v=RstPyqnS9is&list=OLAK5uy_lL-Dp3s8v3NTgVaK82TY81iCn4QElR6Fc&index=9)

Inline `code`

_I am so done with this module already_

0x787af25e committed:
level 2 done

0x787af25e committed:
Merge branch 'master' of github.com:0x787af25e/ip

level 2 done

0x787af25e committed:
done with level 3

0x787af25e committed:
tidy up level 3

0x787af25e committed:
added subclasses for level 4

0x787af25e committed:
started switch statement

0x787af25e committed:
done with deadline, now change arraylists to strings

0x787af25e committed:
done with level 4

0x787af25e committed:
added todo for level 4 (was not completed)

0x787af25e committed:
fixed bug for todo

0x787af25e committed:
done with level 5

0x787af25e committed:
deleted redundant code (level 5)

0x787af25e committed:
done with level 6

0x787af25e committed:
done with auto ui test

0x787af25e committed:
Create method for loading data from file

0x787af25e committed:
Add loading from file

0x787af25e committed:
Added saving to file for delete

0x787af25e committed:
Add time and date functionality

0x787af25e committed:
Merge branch 'branch-Level-8'

Merge level 7 and 8

0x787af25e committed:
Added 4 new classes

0x787af25e committed:
Change main class after adding 4 helper classes

0x787af25e committed:
Added more OOP (barring some bugs)

0x787af25e committed:
Reformat files into package

0x787af25e committed:
Add gradle (and rename package)

0x787af25e committed:
Add Javadocs

0x787af25e committed:
Conform to coding standard

0x787af25e committed:
Update switch statement

0x787af25e committed:
Update commit for JavaDocs

0x787af25e committed:
Added find functionality

0x787af25e committed:
Merge tag 'A-JavaDoc'

0x787af25e committed:
Merge tag 'A-CodingStandard'

Update coding standard (switch statement)

Merge branch-A-CodingStandard, Level-9 and branch-A-JavaDocs

0x787af25e committed:
Add tests

0x787af25e committed:
Add basic UI

bokuanT commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+        this.myTaskList = myTaskList;
+        try {
+            loadFromFile();
+        } catch (IOException ignored) {}
```
Would printing an error message be better than ignoring the `IOException`? Or maybe leave a comment why it is empty

bokuanT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+import duke.*;
```
Perhaps you could list all imported classes explicitly, even if it's the whole package.

bokuanT commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+            String[] toTime = taskAndTime[1].split(" /to ", 2);
+            return new String[]{otherArgs[0], taskAndTime[0], toTime[0], toTime[1]};
+        }
+        return null;
```
Just picking a nit but would you consider using this instead:
```{java}
case EVENT:
    //code
default:
    return null;
```

bokuanT commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,52 @@
+    public boolean getCompletion() {
+        return this.isCompleted;
+    }
+    public void setCompletion() {
```
Perhaps naming this method as `toggleCompletion()` would be better?

0x787af25e replied:
I am mostly sure that the coding standard requires adherence to `get` + the name of the variable for boolean setter methods. Thank you for the other comments though

bokuanT reviewed (commented):
LGTM 👍 

- Good code quality and coding standards adhered to. 

- Just need to fill in javadocs and can consider packaging into more organised packages (I used intelliJ to create new packages and refactor files to automatically package classes)

0x787af25e committed:
Temp commit - need to make Duke say the reply

0x787af25e committed:
Add GUI to Duke (still need to debug)

0x787af25e committed:
Add greeting

kalarisng commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+
+        while ((task = dukeReader.readLine()) != null) {
+            // process it
+            // format: E|0|project meeting|Aug 6th|2pm|4pm
```
i like how clean this piece of code is, thank you for including the format of the saved tasks in the file

kalarisng commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,105 @@
+     */
+    public void printReply(String inputCommand) {
+        switch (inputCommand) {
+            case "list":
```
minor configuration of IDE is required here, as the case clause should not be indented according to the course's java coding standard. however, the rest of the switch cases seem to follow the standard, perhaps you would like to check if this part has been reconfigured?

0x787af25e committed:
Fixed closing app bug

0x787af25e committed:
Fix switch statement indentation

0x787af25e committed:
Add assertions

0x787af25e committed:
Merge pull request #1 from 0x787af25e/branch-A-Assertions

Add assertions

0x787af25e committed:
Improve code quality

Improve code quality

There are many problems with code quality so far in the codebase,
including the lack of else statements, or the use of magic numbers.
These have been replaced with the relevant conventions for better
looking code.

0x787af25e committed:
Merge pull request #2 from 0x787af25e/branch-A-CodeQuality

Improve code quality

0x787af25e committed:
Fix exceptions and bugs

0x787af25e committed:
Add user guide and file
